### PR TITLE
TASK: Carefully fix psalm types across codebase to make it green ;)

### PR DESCRIPTION
**Pull request by mhsdesign**
**Upgrade instructions**

**Review instructions**

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked wit `!!!` and have upgrade-instructions

### CODE REVIEW

**Review by kitsunet — APPROVED** [review]
seems fine

File: Neos.Flow/Classes/Reflection/ReflectionService.php [hunk 1]
```diff
@@ -967,11 +967,12 @@ public function getPropertyAnnotation(string $className, string $propertyName, s
 
     /**
      * Returns the class schema for the given class
+     * @param class-string|string|object $classNameOrObject
      */
     public function getClassSchema(string|object $classNameOrObject): ?ClassSchema
     {
         $className = $classNameOrObject;
-        if (is_object($classNameOrObject)) {
+        if (is_object($className)) {
```
Comment (dlubitz) [inline Neos.Flow/Classes/Reflection/ReflectionService.php]:
```suggestion
        if (is_object($classNameOrObject)) {
```

**Review by dlubitz — APPROVED** [review]

File: Neos.Flow/Classes/Cli/CommandManager.php [hunk 2]
```diff
@@ -270,9 +270,10 @@ public static function getCommandControllerMethodArguments(ObjectManagerInterfac
                 continue;
             }
             $controllerObjectName = $objectManager->getObjectNameByClassName($className);
+            assert($controllerObjectName !== false);
```
Comment (robertlemke) [inline Neos.Flow/Classes/Cli/CommandManager.php]: Maybe rather throw an exception here?

File: Neos.Flow/Classes/Cli/RequestBuilder.php [hunk 3]
```diff
@@ -155,6 +155,7 @@ public function build($commandLine): Request
             return $request;
         }
         $controllerObjectName = $this->objectManager->getObjectNameByClassName($command->getControllerClassName());
+        assert($controllerObjectName !== false);
```
Comment (robertlemke) [inline Neos.Flow/Classes/Cli/RequestBuilder.php]: Anything speaking against throwing an exception instead?

**Review by robertlemke — APPROVED** [review]
Looks good to me generally, but I suggest to throw exceptions instead of using `assert()` in the mentioned cases.

File: Neos.Flow/Classes/Cli/CommandManager.php (hunk 2 above)
Comment (mhsdesign) [reply Neos.Flow/Classes/Cli/CommandManager.php]: adjusted

File: Neos.Flow/Classes/Cli/RequestBuilder.php (hunk 3 above)
Comment (mhsdesign) [reply Neos.Flow/Classes/Cli/RequestBuilder.php]: adjusted

**Pull request merged**